lib/operations/create_indexes.js

```diff
@@ -41,21 +47,30 @@ class CreateIndexesOperation extends OperationBase {
       }
     }
 
-    options = Object.assign({}, options, { readPreference: ReadPreference.PRIMARY });
-
-    // Execute the index
-    executeCommand(
-      coll.s.db,
-      {
-        createIndexes: coll.collectionName,
-        indexes: indexSpecs
-      },
-      options,
-      callback
-    );
+    const cmd = {
+      createIndexes: this.name,
+      indexes: indexSpecs
+    };
+
+    if (options.commitQuorum != null) {
+      // Ensure commitQuorum not passed if server < 4.4
+      if (commitQuorumNotSupported) {
+        return callback(new MongoError(commitQuorumNotSupported));
+      }
+      cmd.commitQuorum = options.commitQuorum;
+    }
+
+    super.executeCommand(server, cmd, (err, result) => {
+      if (err) {
+        callback(err);
+        return;
+      }
+
+      callback(null, this.singular ? indexSpecs[0].name : result);
```

Comment:
Added `this.singular` because the `createIndex` and `createIndexes` have return different things.

Reply:
> Ideally, we would not do anything different for singular vs plural, just return the server response unmolested but I think that might be a breaking change here. If so, let's add a breaking change for Node v4

That's the issue, `createIndex` just returns the name of the created index on success, whereas `createIndexes` returns the entire response.